tests/phpunit/Civi/Funding/FundingCase/FundingCaseManagerTest.php

```diff
@@ -165,31 +167,57 @@ public function testGetOpenOrCreate(): void {
     });
 
     RequestTestUtil::mockInternalRequest($contact['id']);
-    $fundingCase = $this->fundingCaseManager->getOpenOrCreate($contact['id'], [
+    $fundingCase = $this->fundingCaseManager->getOrCreate($contact['id'], [
       'funding_program' => $fundingProgram,
       'funding_case_type' => $fundingCaseType,
       'recipient_contact_id' => $recipientContact['id'],
       'title' => 'Title',
-    ]);
-
+    ], []);
     static::assertSame('open', $fundingCase->getStatus());
-    static::assertEquals($fundingCase, $this->fundingCaseManager->getOpenOrCreate($contact['id'], [
+
+    $fundingCaseSame = $this->fundingCaseManager->getOrCreate($contact['id'], [
       'funding_program' => $fundingProgram,
       'funding_case_type' => $fundingCaseType,
       'recipient_contact_id' => $recipientContact['id'],
       'title' => 'Title',
-    ]));
+    ], ['open']);
+    static::assertEquals($fundingCase, $fundingCaseSame);
 
-    $fundingCase->setStatus('test');
-    FundingCase::update()->setValues($fundingCase->toArray())->execute();
+    // Current status not in allowed existing status.
+    $fundingCase2 = $this->fundingCaseManager->getOrCreate($contact['id'], [
+      'funding_program' => $fundingProgram,
+      'funding_case_type' => $fundingCaseType,
+      'recipient_contact_id' => $recipientContact['id'],
+      'title' => 'Title',
+    ], ['foo']);
+    static::assertNotSame($fundingCase->getId(), $fundingCase2->getId());
 
-    $fundingCase2 = $this->fundingCaseManager->getOpenOrCreate($contact['id'], [
+    // Empty allowed existing status.
+    $fundingCase3 = $this->fundingCaseManager->getOrCreate($contact['id'], [
       'funding_program' => $fundingProgram,
       'funding_case_type' => $fundingCaseType,
       'recipient_contact_id' => $recipientContact['id'],
       'title' => 'Title2',
-    ]);
-    static::assertNotSame($fundingCase->getId(), $fundingCase2->getId());
+    ], []);
+    static::assertNotSame($fundingCase->getId(), $fundingCase3->getId());
+
+    // Different funding program.
+    $fundingCase4 = $this->fundingCaseManager->getOrCreate($contact['id'], [
+      'funding_program' => $fundingProgram2,
+      'funding_case_type' => $fundingCaseType,
+      'recipient_contact_id' => $recipientContact['id'],
+      'title' => 'Title',
+    ], ['open']);
+    static::assertNotSame($fundingCase->getId(), $fundingCase4->getId());
```

Comment:
for understanding the test:

if a funding case is being created, does it has the status `open`?
if so, then `$fundingCase` would be in status `open` as well.
should `getOrCreate` then return  `$fundingCase` here instead of creating a new funding-case?
thus would `$fundingCase` and `$fundingCase4` be identical?